src/core/thread/mle.cpp

```diff
@@ -246,6 +252,7 @@ ThreadError Mle::Stop(void)
         mNetif.RemoveUnicastAddress(mLeaderAloc);
     }
 
+    mNetif.GetPendingDataset().StopDelayTimer();
```

Comment:
Where do we restart the Delay Timer?  If we detach and we have a Pending Dataset, we need to make sure that we apply the Pending Dataset after the Delay Timer expires.